src/ai/ai_gui.cpp

```diff
@@ -180,17 +180,17 @@ struct AIConfigWindow : public Window {
 
 					if ((_game_mode != GM_NORMAL && i == 0) || (_game_mode == GM_NORMAL && Company::IsValidHumanID(i))) {
 						text = STR_AI_CONFIG_HUMAN_PLAYER;
-					} else if (AIConfig::GetConfig((CompanyID)i)->GetInfo() != nullptr) {
-						SetDParamStr(0, AIConfig::GetConfig((CompanyID)i)->GetInfo()->GetName());
+					} else if (AIConfig::GetConfig(static_cast<CompanyID>(i))->GetInfo() != nullptr) {
+						SetDParamStr(0, AIConfig::GetConfig(static_cast<CompanyID>(i))->GetInfo()->GetName());
 						text = STR_JUST_RAW_STRING;
 					} else {
 						text = STR_AI_CONFIG_RANDOM_AI;
 					}
 
 					TextColour tc = TC_SILVER;
-					if (this->selected_slot == i) {
+					if (this->selected_slot == static_cast<CompanyID>(i)) {
 						tc = TC_WHITE;
-					} else if (IsEditable((CompanyID)i)) {
+					} else if (IsEditable(static_cast<CompanyID>(i))) {
```

Comment:
This is probably at the point where I'd put `CompanyID cid = static_cast<CompanyID>(i);` at the beginning of the loop, and replace all those casts.